internal/stats/course.go

```diff
@@ -33,16 +34,6 @@ type CourseMetric struct {
 	Value uint64 `json:"duration"`
 }
 
-type CourseMetricQuery struct {
```

Comment:
Since you are making simplification to the query like this,
would it make sense to also simplify the metric to just use `BaseMetric.Attributes` instead of fields.
Then you wouldn't have to deal with matching JSON fields.

Note that this is not an order or strong stance.
This is something for you to really think about and decide if it would be better for your code.